core/services/keeper/upkeep_executer.go

```diff
@@ -278,7 +278,7 @@ func (ex *UpkeepExecuter) estimateGasPrice(upkeep UpkeepRegistration) (gasPrice
 	}
 
 	if ex.config.EvmEIP1559DynamicFees() {
-		fee, _, err = ex.gasEstimator.GetDynamicFee(upkeep.ExecuteGas)
+		fee, _, err = ex.gasEstimator.GetDynamicFee(upkeep.ExecuteGas, ex.config.KeySpecificMaxGasPriceWei(upkeep.Registry.FromAddress.Address()))
```

Comment:
👍 

not sure if it's easy to abstract the key specific limit inside gasEstimator so that callers don't have to worry about it. would be cleaner if possible

Reply:
yeah I tried something similar and the clients have different config interfaces so it seemed less extensible to pass configs directly

Reply:
I think on balance its nicer to pass in the max gas price rather than have the estimator be aware of the key map.

This is because we may introduce other ways to limit gas in future and its nice to be extensible for that.